p2-maven-plugin/src/main/java/org/eclipse/tycho/p2maven/transport/RemoteArtifactRepositoryManagerAgentFactory.java

```diff
@@ -45,14 +49,41 @@ public Object createService(IProvisioningAgent agent) {
 	}
 
 	private boolean getDisableP2MirrorsConfiguration() {
-		String key = "tycho.disableP2Mirrors";
-		String value = System.getProperty(key);
-		if (value != null) {
-			logger.info("Using " + key
+		String deprecatedKey = "tycho.disableP2Mirrors";
+		String deprecatedValue = System.getProperty(deprecatedKey);
+
+		if (deprecatedValue == null) {
+			deprecatedValue = mavenContext.getSession().getSystemProperties().getProperty(deprecatedKey);
```

Comment:
The session might be null so you should extract it and make sure it is only used when not null.